tests/cunit/test_darray.c

```diff
@@ -0,0 +1,317 @@
+    /* /\* if (data_in != my_rank * 10) *\/ */
+    /* /\*     return ERR_WRONG; *\/ */
+
+    /* /\* Close the file. *\/ */
+    /* if ((ret = PIOc_closefile(ncid))) */
+    /*     return ret; */
+
+    /* /\* Free the PIO decomposition. *\/ */
+    /* if ((ret = PIOc_freedecomp(iosysid, ioid))) */
+    /*     ERR(ret); */
+
+    return PIO_NOERR;
+}
+
+/** 
+ * Test the darray functionality. Create a netCDF file with 3
+ * dimensions and 1 variable, and use darray to write some data.
+ *
+ * @param iosysid the IO system ID.
+ * @param ioid the ID of the decomposition.
+ * @param num_flavors the number of IOTYPES available in this build.
+ * @param flavor array of available iotypes.
+ * @param my_rank rank of this task.
+ * @returns 0 for success, error code otherwise.
+*/
+int test_darray(int iosysid, int ioid, int num_flavors, int *flavor, int my_rank)
+{
+    char filename[PIO_MAX_NAME + 1]; /* Name for the output files. */
+    int dimids[NDIM];      /* The dimension IDs. */
+    int ncid;      /* The ncid of the netCDF file. */
+    int varid;     /* The ID of the netCDF varable. */
+    int ret;       /* Return code. */
+
+    /* Use PIO to create the example file in each of the four
+     * available ways. */
+    /* for (int fmt = 0; fmt < num_flavors; fmt++) */
+    for (int fmt = 1; fmt < 2; fmt++)
+    {
+        /* Create the filename. */
+        sprintf(filename, "data_%s_iotype_%d.nc", TEST_NAME, flavor[fmt]);
+
+        /* Create the netCDF output file. */
+        printf("rank: %d Creating sample file %s with format %d...\n", my_rank, filename,
+               flavor[fmt]);
+        if ((ret = PIOc_createfile(iosysid, &ncid, &flavor[fmt], filename, PIO_CLOBBER)))
+            ERR(ret);
+
+        /* Define netCDF dimensions and variable. */
+        printf("rank: %d Defining netCDF metadata...\n", my_rank);
+        for (int d = 0; d < NDIM; d++)
+            if ((ret = PIOc_def_dim(ncid, dim_name[d], (PIO_Offset)dim_len[d], &dimids[d])))
+                ERR(ret);
+
+        /* Define a variable. */
+        if ((ret = PIOc_def_var(ncid, VAR_NAME, PIO_INT, NDIM, dimids, &varid)))
+            ERR(ret);
+
+        /* End define mode. */
+        if ((ret = PIOc_enddef(ncid)))
+            ERR(ret);
+
+        /* Write some data. */
+        PIO_Offset arraylen = 4;
+        int fillvalue = NC_FILL_INT;
+        int test_data[arraylen];
+        for (int f = 0; f < arraylen; f++)
+            test_data[f] = my_rank * 10 + f;
+
+        if ((ret = PIOc_setframe(ncid, varid, 0)))
+            ERR(ret);
+        
+        if ((ret = PIOc_write_darray(ncid, varid, ioid, arraylen, test_data, &fillvalue)))
+            ERR(ret);
+
+        /* Close the netCDF file. */
+        printf("rank: %d Closing the sample data file...\n", my_rank);
+        if ((ret = PIOc_closefile(ncid)))
+            ERR(ret);
+
+        /* Check the file contents. */
+        /* if ((ret = check_darray_file(iosysid, TARGET_NTASKS, my_rank, filename))) */
```

Comment:
Is checking the contents slated for a future PR? May be you can remove this function (check_darray_file()) for now.